hack/run-go-test.sh

```diff
@@ -0,0 +1,17 @@
+#! /bin/bash
```

Comment:
`run-go-test.sh` is a bit too vague for what this file does. Can you pick a new name that's more representative like `run-e2e.sh` or something. Also it seems like this script could just be inlined into `e2e.sh`, not really sure we need it to be a separate script.

Reply:
The only reason that this needs to be its own standalone script is to capture the go test flags used.

I agree the name was poorly chosen and unfortunately `run-e2e-tests.sh` would be the more descriptive name. I'll update the name, and maybe we can rename this after integration is refactored too.

Reply:
Not sure I understand the "to capture the flags used" part. Can you elaborate?

Reply:
@chancez if inlined the multi-line go test command into `./hack/e2e.sh`, then the `tee` command doesn't log any of the multi-line go test command (so none of the values passed to the go test flags are logged), which seems like poor behavior.

Reply:
More concretely:
```bash
└─ $ ▶ bat /tmp/tmp.YDEJch1xed/metering-tflannag-e2e/tests/e2e-tests.log 
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /tmp/tmp.YDEJch1xed/metering-tflannag-e2e/tests/e2e-tests.log
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ time="10-14-2019 18:04:42" level=info msg="Setting the log level to debug" app=deploy
   2   │ === RUN   TestMultipleInstalls
   3   │ === RUN   TestMultipleInstalls/HDFSInstall
   4   │ time="10-14-2019 18:04:42" level=debug msg="Deployer config: {SkipMeteringDeployment:false DeleteCRDs:false DeleteCRB:false DeleteNamespace:true DeletePVCs:false DeleteAll:false Namespace:metering-tflannag-e2e-7 Platform:openshift Repo: Tag: OperatorResources:0xc000137900 MeteringConfig:0xc00000d2c0}" app=de
       │ ploy
   5   │ time="10-14-2019 18:04:42" level=info msg="Metering Deploy Namespace: metering-tflannag-e2e-7" app=deploy
   6   │ time="10-14-2019 18:04:42" level=info msg="Metering Deploy Platform: openshift" app=deploy
   7   │ time="10-14-2019 18:04:42" level=info msg="Labeling the metering-tflannag-e2e-7 namespace with 'openshift.io/cluster-monitoring=true'" app=deploy
   8   │ time="10-14-2019 18:04:42" level=info msg="Created the metering-tflannag-e2e-7 namespace" app=deploy
   9   │ time="10-14-2019 18:04:42" level=info msg="Updated the hivetables.metering.openshift.io CRD" app=deploy
  10   │ time="10-14-2019 18:04:42" level=info msg="Updated the prestotables.metering.openshift.io CRD" app=deploy
  11   │ time="10-14-2019 18:04:42" level=info msg="Updated the storagelocations.metering.openshift.io CRD" app=deploy
  12   │ time="10-14-2019 18:04:42" level=info msg="Updated the reports.metering.openshift.io CRD" app=deploy
  13   │ time="10-14-2019 18:04:42" level=info msg="Updated the reportqueries.metering.openshift.io CRD" app=deploy
  14   │ time="10-14-2019 18:04:42" level=info msg="Updated the reportdatasources.metering.openshift.io CRD" app=deploy
  15   │ time="10-14-2019 18:04:43" level=info msg="Updated the meteringconfigs.metering.openshift.io CRD" app=deploy
  16   │ time="10-14-2019 18:04:43" level=info msg="Created the metering deployment" app=deploy
  17   │ time="10-14-2019 18:04:43" level=info msg="Created the metering serviceaccount" app=deploy
  18   │ time="10-14-2019 18:04:43" level=info msg="Created the metering role" app=deploy
  19   │ time="10-14-2019 18:04:43" level=info msg="Created the metering role binding" app=deploy
  20   │ time="10-14-2019 18:04:43" level=info msg="Created the metering cluster role" app=deploy
  21   │ time="10-14-2019 18:04:43" level=info msg="Created the metering cluster role binding" app=deploy
  22   │ time="10-14-2019 18:04:43" level=info msg="Created the MeteringConfig resource" app=deploy
  23   │ time="10-14-2019 18:04:43" level=info msg="Updated the 'name=metering-testing-ns' label to the metering-tflannag-e2e-7 namespace" app=deploy
  24   │ time="10-14-2019 18:04:43" level=info msg="Waiting for all metering pods to be ready" app=deploy
  25   │ time="10-14-2019 18:04:53" level=info msg="Poll Summary" app=deploy
  26   │ time="10-14-2019 18:04:53" level=info msg="Current ratio of ready to target pods: 1/7" app=deploy
  27   │ time="10-14-2019 18:05:03" level=info msg="Poll Summary" app=deploy
  28   │ time="10-14-2019 18:05:03" level=info msg="Current ratio of ready to target pods: 1/7" app=deploy
```

Reply:
In comparison to what's current in this PR:
```bash
└─ $ ▶ bat /tmp/tmp.jP958XLTky/metering-tflannag-e2e/tests/e2e-tests.log 
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /tmp/tmp.jP958XLTky/metering-tflannag-e2e/tests/e2e-tests.log
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ + go test -test.short=false -test.v -parallel 10 -timeout 30m /home/tflannag/go/src/github.com/operator-framework/operator-metering/test/e2e -kubeconfig=/home/tflannag/openshift-dev/auth/kubeconfig -namespace-prefix=metering-tflannag-e2e -deploy-manifests-dir=/home/tflannag/go/src/github.com/operator-framewo
       │ rk/operator-metering/manifests/deploy -cleanup-script-path=/home/tflannag/go/src/github.com/operator-framework/operator-metering/hack/run-test-cleanup.sh -test-output-path=/tmp/tmp.jP958XLTky/metering-tflannag-e2e -log-level=debug
   2   │ time="10-14-2019 18:06:53" level=info msg="Setting the log level to debug" app=deploy
   3   │ === RUN   TestMultipleInstalls
   4   │ === RUN   TestMultipleInstalls/HDFSInstall
   5   │ time="10-14-2019 18:06:53" level=debug msg="Deployer config: {SkipMeteringDeployment:false DeleteCRDs:false DeleteCRB:false DeleteNamespace:true DeletePVCs:false DeleteAll:false Namespace:metering-tflannag-e2e-38 Platform:openshift Repo: Tag: OperatorResources:0xc0002e19a0 MeteringConfig:0xc00000cf00}" app=d
       │ eploy
```

Reply:
That can be accomplished by putting `set -x` before running Go test.

Reply:
I originally had `set -x` set before running the `go test ...` portion, and still wasn't able to get that to work. 

Maybe I need to do something different with the file descriptors, but I've tried a couple of different combinations and haven't been able to find any references that achieve this behavior.

Reply:
`set -x` is what does this, and it doesn't need a separate file for it to work, it's just printing the shell expressions as they're executed. It doesn't have anything to do with files or file descriptors.

Reply:
I was under the impresison that `set -x` writes to stderr (which is a file descriptor).

Reply:
Ah, that might be true, though I don't necessarily think that's the issue here. I do see in `e2e.sh` you put `set +x` though.